velox/experimental/cudf-exchange/Communicator.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <ucxx/api.h>
+#include <cstdint>
+#include <string>
+#include "velox/experimental/cudf-exchange/Acceptor.h"
+#include "velox/experimental/cudf-exchange/CommElement.h"
+#include "velox/experimental/cudf-exchange/WorkQueue.h"
+
+namespace facebook::velox::cudf_exchange {
+
+struct HostPort {
+  std::string hostname;
+  uint16_t port = 0;
+
+  // Default and parameterized constructors
+  HostPort() = default;
+  HostPort(std::string h, uint16_t p) : hostname(std::move(h)), port(p) {}
+
+  // Strict weak ordering for std::map
+  bool operator<(HostPort const& other) const noexcept {
+    if (hostname < other.hostname)
+      return true;
+    if (other.hostname < hostname)
+      return false;
+    return port < other.port;
+  }
+};
+
+class Communicator {
+  friend Acceptor;
+
+ public:
+  const ucxx::AmReceiverCallbackOwnerType kAmCallbackOwner = "communicator";
```

Comment:
```suggestion
  const ucxx::AmReceiverCallbackOwnerType kAmCallbackOwner = "velox";
```

Reply:
Not of big immediate concern, the owner string is expected to be something relatively unique that's unlikely for a different application to use as well and cause a collision, so `"communicator"` may be too generic, and the name of the application is not.